src/move/econia/sources/market.move

```diff
@@ -3077,6 +3078,31 @@ module econia::market {
             let order_ref_mut = avl_queue::borrow_head_mut(orders_ref_mut);
             // Assert AVL queue head price matches that of order.
             assert!(order_ref_mut.price == price, E_HEAD_KEY_PRICE_MISMATCH);
+            // If order at head of queue has size 0, evict it and
+            // continue to next order. This should never be reached
+            // during production, but is handled here to explicitly
+            // verify the assumption of no empty orders on the book.
+            if (order_ref_mut.size == 0) {
+                let Order{
+                    size: evictee_size,
+                    price: evictee_price,
+                    user: evictee_user,
+                    custodian_id: evictee_custodian_id,
+                    order_access_key: evictee_order_access_key
+                } = avl_queue::pop_head(orders_ref_mut);
+                user::cancel_order_internal(
```

Comment:
Should we emit a fill event? There was no amount filled but we did match and remove an order from the book, which might be important to off-chain API's that are expecting either a fill or a cancel to remove the order from the book.